### DISCUSSION TITLE
How would you change style depending on Breakpoints ?

### DISCUSSION THREAD

**Discussion by n44ps** [mantinedev/mantine#338] [Q&A] [answered]
Hello !

I am new to Mantine, searched the doc, but I cannot see a way to style Components with an array-like value to some props. Like styled-system would use :

```tsx
<Flex flexDirection={['column', 'column', 'row']}>...</Flex>
```

I see we have `useMediaQuery`, but how does it compare to the array-like way ? Is it more performant over dx ?

```tsx
export function Demo() {
  const matches = useMediaQuery('(min-width: 900px)');

  return (
    <Badge color={matches ? 'teal' : 'red'} variant="filled">
      Breakpoint {matches ? 'matches' : 'does not match'}
    </Badge>
  );
}
```

Thank you :)

**Comment by rtivital** [comment, marked as answer]
Hi @n44ps, you can use `createStyles` function to create responsive styles with media queries – https://mantine.dev/theming/create-styles/#media-queries

**Reply by n44ps** [reply]
Hey @rtivital, thanks for the answer :)

Have a great weekend !

**Comment by ilomon10** [comment]
How to make `variant` responsive?
```tsx
<Button variant={["filled","outline"]}>
  ...
</Button>
```

**Reply by rtivital** [reply]
https://mantine.dev/theming/responsive/#mediaquery-component

**Reply by ilomon10** [reply]
just applied like this
``` jsx
...
const { theme } = useMantineTheme();
const mediaQuery = useMediaQuery(`(max-width: ${theme.breakpoints.md}px)`);
return (
  <Tabs variant={mediaQuery ? "pills" : "default"}>
    ...
  </Tabs>
)
...
```

and it shows warning 
```console
Warning: Prop `className` did not match. Server: ...
```

**Reply by rtivital** [reply]
useMediaquery hook works only in browser, it will have className mismatch every time

**Reply by ilomon10** [reply]
It's on browser but first it SSR then i resize my window.
I use NextJS.

*maybe when in production no one can resize their phone or screen

Thanks for replying

**Comment by kaisiant** [comment]
Is there any other way to make the component responsive? Coz with Chakra UI, it is so simple.

Chakra:

```
<Box bg='red.200' w={[300, 400, 500]}>
  This is a box
</Box>

<Text fontSize={{ base: '24px', md: '40px', lg: '56px' }}>
  This is responsive text
</Text>
```
Mantine:

```
<TextInput
   sx={(theme) => ({
     background: theme.colors.gray[0],
     padding: theme.spacing.md,

     '@media (max-width: 40em)': {
       padding: theme.spacing.sm,
     },
   })}
 />
```

**Reply by rtivital** [reply]
https://mantine.dev/styles/style-props/

**Comment by rj-wowza** [comment]
> but how does it compare to the array-like way ? Is it more performant over dx ?

I am also curious as to why Mantine is so verbose when it comes to accessing theme values and creating responsive styles. Is there a performance or scalability concern, or is it a matter of someone contributing the necessary code?
